pkg/privilege/privileges/cache.go

```diff
@@ -506,78 +572,87 @@ func (p *immutable) loadSomeUsers(ctx sqlexec.RestrictedSQLExecutor, userList ..
 }
 
 // merge construct a new MySQLPrivilege by merging the data of the two objects;.
-func (p *MySQLPrivilege) merge(diff *immutable, userList []string) *MySQLPrivilege {
-	var ret MySQLPrivilege
-	ret.user = make([]UserRecord, 0, len(p.user)+len(diff.user))
-	for _, v := range p.user {
-		if !slices.ContainsFunc(userList, func(usr string) bool { return usr == v.User }) {
-			ret.user = append(ret.user, v)
-		}
-	}
-	ret.user = append(ret.user, diff.user...)
-	slices.SortFunc(ret.user, compareUserRecord)
-	ret.buildUserMap()
+func (p *MySQLPrivilege) merge(diff *MySQLPrivilege, userList []string) *MySQLPrivilege {
+	ret := newMySQLPrivilege()
+	user := p.user.Clone()
```

Comment:
Do we need some mutex for `Clone` ? The BTreeG is documented as:

> // Clone clones the btree, lazily.  Clone should not be called concurrently,
// but the original tree (t) and the new tree (t2) can be used concurrently
// once the Clone call completes.

Reply:
Done